pilot/pkg/config/kube/crd/client.go

```diff
@@ -47,9 +47,11 @@ import (
 
 // IstioAPIGroupVersion defines schema.GroupVersion for Istio configuration
 // resources.
+// TODO(xiaolanz) Client can only write to one apiVersion until https://github.com/istio/istio/issues/3586
+// is fully completed.
 var IstioAPIGroupVersion = schema.GroupVersion{
-	Group:   model.IstioAPIGroup,
-	Version: model.IstioAPIVersion,
+	Group:   "config.istio.io",
```

Comment:
can you make this a model.IstioDefaultAPIGroup and ditto for version rather than hardcode this here
despite the TODO that doesn't seem like an improvement

Reply:
I intentionally not to introduce something like model.IstioDefaultAPIGroup. Same for the version. Everything out of model package should resolve group/version from schema dynamically.

IstioAPIGroupVersion together with the strings will be gone because I plan to do client change in a separate PR.